lang/cs/Org.Apache.REEF.IO/FileSystem/AzureDataLake/AzureDataLakeFileSystem.cs

```diff
@@ -0,0 +1,235 @@
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new AdlsException(ex.Message);
+            }
+        }
+
+        /// <summary>
+        /// Copies the specified file to the remote location.
+        /// </summary>
+        /// <exception cref="AdlsException">If copy process encounters any exception</exception>
+        public void CopyFromLocal(string localFileName, Uri remoteFileUri)
+        {
+            try
+            {
+                TransferStatus status = _adlsClient.BulkUpload(localFileName, remoteFileUri.AbsolutePath);
+                if (status.EntriesFailed.Count != 0)
+                {
+                    throw new AdlsException($"{status.EntriesFailed.Count} entries did not get transferred correctly");
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new AdlsException(ex.Message);
+            }
+        }
+
+        /// <summary>
+        /// Creates a new directory.
+        /// </summary>
+        /// <exception cref="AdlsException">If directory cannot be created</exception>
+        public void CreateDirectory(Uri directoryUri)
+        {
+            bool createDirStatus = _adlsClient.CreateDirectory(directoryUri.AbsolutePath);
+            if (!createDirStatus)
+            {
+                throw new AdlsException($"Cannot create directory specified by {directoryUri.ToString()}");
+            }
+        }
+
+        /// <summary>
+        /// Deletes a directory.
+        /// </summary>
+        /// <exception cref="AdlsException">If directory cannot be deleted</exception>
+        public void DeleteDirectory(Uri directoryUri)
+        {
+            bool deleteStatus = Exists(directoryUri) && 
+                _adlsClient.GetDirectoryEntry(directoryUri.AbsolutePath).Type == DirectoryEntryType.DIRECTORY &&
+                _adlsClient.DeleteRecursive(directoryUri.AbsolutePath);
+            if (!deleteStatus)
+            {
+                throw new AdlsException($"Cannot delete directory specified by {directoryUri.ToString()}");
+            }
+        }
+
+        /// <summary>
+        /// Get the children on the given URI, if that refers to a directory.
+        /// </summary>
+        /// <exception cref="AdlsException">If directory does not exist</exception>
+        public IEnumerable<Uri> GetChildren(Uri directoryUri)
+        {
+            if (Exists(directoryUri) && _adlsClient.GetDirectoryEntry(directoryUri.AbsolutePath).Type == DirectoryEntryType.DIRECTORY)
+            {
+                foreach (var entry in _adlsClient.EnumerateDirectory(directoryUri.AbsolutePath))
+                {
+                    yield return new Uri($"{GetUriPrefix()}{entry.FullName}");
+                }
+            }
+            else
+            {
+                throw new AdlsException($"Cannot find directory specified by {directoryUri.ToString()}");
```

Comment:
Maybe reverse the condition for slightly shorter code (i.e. no `else` block)?
```C#
if (!Exists(directoryUri) ||
    _adlsClient.GetDirectoryEntry(directoryUri.AbsolutePath).Type != DirectoryEntryType.DIRECTORY)
{
    throw new AdlsException($"Cannot find directory specified by {directoryUri}");
}

return _adlsClient.EnumerateDirectory(directoryUri.AbsolutePath)
    .Select(entry => new Uri($"{GetUriPrefix()}{entry.FullName}"));
```

Reply:
done